### ISSUE TITLE
Uploader has broken features

### ISSUE THREAD

**Issue by tonypee** [ElemeFE/element#3256]
Im having some issues with the uploader not functioning as expected - It seems 

Returning 'false' from 'before-upload' - doesnt stop the upload.

:multiple="false" - doesnt restrict multiple uploads

accept="text/csv" - doesnt restrict the type

I am using it based on the drag-and-drop example, and the results just seem unexpected:

```
<el-upload
              class="upload-demo"
              drag
              accept="text/csv"
              action="//jsonplaceholder.typicode.com/posts/"
              :on-success="handleSuccess"
              :on-preview="handlePreview"
              :on-remove="handleRemove"
              :before-upload="restrictUpload"
              :multiple="false"
              :file-list="fileList">
              <i class="el-icon-upload"></i>
              <div class="el-upload__text">Dropa file here or <em>click to upload</em></div>
              <div class="el-upload__tip" slot="tip">jpg/png files with a size less than 500kb</div>
            </el-upload>


    handleSuccess(v) {
      console.log('success', v)
    }

    handlePreview(v) {
      console.log('preview', v)
    }

    handleRemove(v) {
      console.log('remove', v)
    }

    restrictUpload(file) {
      console.log('kill')
      return false
   }
```

the 'kill' will fire for example, but the upload still continues. and i can add multiple

**Comment by Leopoldthecoder** [comment]
I'm afraid you're not using the latest version of Element. A working demo: http://jsfiddle.net/leopoldthecuber/zj305kno/1/

Just to clarify, `multiple` means if you can select multiple files in the file-selection popup at one time. It doesn't mean if you can upload several files to the server.

**Comment by 188007803** [comment]
第一个问题，还是没有解决啊，在 before-upload 事件中返回为 false，文件名还是传到文件列表中了，应该不出现在文件列表中。

**Comment by baiyaaaaa** [comment]
fix in #3609

**Comment by ThisIsntMyId** [comment]
the accept is still not working
